rosbag2_transport/src/rosbag2_transport/rosbag2_transport_python.cpp

```diff
@@ -128,7 +163,23 @@ rosbag2_transport_play(PyObject * Py_UNUSED(self), PyObject * args, PyObject * k
   play_options.node_prefix = std::string(node_prefix);
   play_options.read_ahead_queue_size = read_ahead_queue_size;
 
-  rosbag2_transport::Rosbag2Transport transport;
+  rosbag2_storage::MetadataIo metadata_io{};
+  rosbag2_storage::BagMetadata metadata{};
+  // Specify defaults
+  auto info = std::make_shared<rosbag2_cpp::Info>();
+  auto reader = std::make_shared<rosbag2_cpp::Reader>(
+    std::make_unique<rosbag2_cpp::readers::SequentialReader>());
+  auto writer = std::make_shared<rosbag2_cpp::Writer>(
+    std::make_unique<rosbag2_cpp::writers::SequentialWriter>());
```

Comment:
These should be declared only since the following if-block may reinitialize the variables.